### PR TITLE
[#46] [Integrate] As a user, I can navigate to the next question

### LINKED ISSUE #46: [Integrate] As a user, I can navigate to the next question
## Why

In the Survey Question screen, the user can navigate to the next question when they finish the current question

## Acceptance Criteria

- When tapping the Arrow Button
    - Store the answer
    - Navigate to the next question
- If the question is the last, the Arrow Button needs to be replaced with 
CleanShot 2023-04-07 at 15 04 40

## Design

CleanShot 2023-04-07 at 15 02 37
Link: https://www.figma.com/file/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=2632-27&t=uQO5aspdvq0F8aof-4

### PR DESCRIPTION
**Pull request by markgravity**
https://github.com/markgravity/kmm-ic/issues/46

## What happened 👀

- Refactor answer views to set the answer to `SurveyQuestionViewModel`
- Replace `ArrowButton` with `SubmitButton` when in the last question
- Write more tests for `SurveyQuestionViewModel`

## Insight 📝

We will integrate the Submit button in #39 

## Proof Of Work 📹

https://github.com/markgravity/kmm-ic/assets/17875522/106baf12-c25a-4849-91d3-9dcc726ff9e6

### CODE REVIEW

File: ios/KMMICTests/Sources/Specs/Presentation/Modules/SurveyQuestion/SurveyQuestionViewModelSpec.swift [hunk 1]
```diff
@@ -35,9 +35,166 @@ final class SurveyQuestionViewModelSpec: QuickSpec {
...
+
+            describe("its isAllValid") {
+                let displayTypes: [QuestionDisplayType] = [.dropdown, .heart, .star, .smiley, .nps, .textarea, .choice]
+                for displayType in displayTypes {
+                    let surveyDetail = SurveyDetail.dummy(questions: [
+                        .dummy(id: "1", displayType: displayType)
+                    ])
+                    context("when displayType is \(displayType)") {
+
+                        context("when answer inputs is empty") {
```
Comment (phongvhd93) [inline ios/KMMICTests/Sources/Specs/Presentation/Modules/SurveyQuestion/SurveyQuestionViewModelSpec.swift]: when answer inputs are empty

File: ios/KMMICTests/Sources/Specs/Presentation/Modules/SurveyQuestion/SurveyQuestionViewModelSpec.swift [hunk 2]
```diff
@@ -35,9 +35,166 @@ final class SurveyQuestionViewModelSpec: QuickSpec {
...
+                                    questionIndex: 0
+                                )
+                            }
+
+                            it("returns isAllValid correctly") {
+                                expect(viewModel.isAllValid) == false
+                            }
+                        }
+
+                        context("when answer inputs is not empty") {
```
Comment (phongvhd93) [inline ios/KMMICTests/Sources/Specs/Presentation/Modules/SurveyQuestion/SurveyQuestionViewModelSpec.swift]: when answer inputs are not empty

**Review by phongvhd93 — COMMENTED** [review]
Only a few minor grammar errors; please check for similar cases. The rest looks good.

File: ios/KMMICTests/Sources/Specs/Presentation/Modules/SurveyQuestion/SurveyQuestionViewModelSpec.swift (hunk 2 above)
Comment (markgravity) [reply ios/KMMICTests/Sources/Specs/Presentation/Modules/SurveyQuestion/SurveyQuestionViewModelSpec.swift]: @phongvhd93 updated at [fef4e7f](https://github.com/markgravity/kmm-ic/pull/96/commits/fef4e7fbe8f7bf595a52de289ebe31252be2e656)

File: ios/KMMICTests/Sources/Specs/Presentation/Modules/SurveyQuestion/SurveyQuestionViewModelSpec.swift (hunk 1 above)
Comment (markgravity) [reply ios/KMMICTests/Sources/Specs/Presentation/Modules/SurveyQuestion/SurveyQuestionViewModelSpec.swift]: @phongvhd93 updated at [fef4e7f](https://github.com/markgravity/kmm-ic/pull/96/commits/fef4e7fbe8f7bf595a52de289ebe31252be2e656)

**Review by phongvhd93 — APPROVED** [review]

**Review by suho — APPROVED** [review]

**Pull request merged**